app/stores/ada/LedgerSendStore.js

```diff
@@ -115,41 +112,36 @@ export default class LedgerSendStore extends Store {
       this._setError(null);
       this._setActionProcessing(true);
 
-      if (this.ledgerBridge) {
-        // Since this.ledgerBridge is undefinable flow need to know that it's a LedgerBridge
-        const ledgerBridge: LedgerBridge = this.ledgerBridge;
-
-        const stateFetcher = this.stores.substores[environment.API].stateFetchStore.fetcher;
-        this.createLedgerSignTxDataRequest.execute({
-          ...params,
-          getUTXOsForAddresses: stateFetcher.getUTXOsForAddresses,
-          getTxsBodiesForUTXOs: stateFetcher.getTxsBodiesForUTXOs,
-        });
-        if (!this.createLedgerSignTxDataRequest.promise) throw new Error('should never happen');
-        const ledgerSignTxDataResp = await this.createLedgerSignTxDataRequest.promise;
-
-        await prepareLedgerBridger(ledgerBridge);
-
-        const ledgerSignTxResp: LedgerSignTxResponse =
-          await ledgerBridge.signTransaction(
-            ledgerSignTxDataResp.ledgerSignTxPayload.inputs,
-            ledgerSignTxDataResp.ledgerSignTxPayload.outputs,
-          );
-
-        await this._prepareAndBroadcastSignedTx(
-          ledgerSignTxResp,
-          params.signRequest.unsignedTx,
+
+      const stateFetcher = this.stores.substores[environment.API].stateFetchStore.fetcher;
+      this.createLedgerSignTxDataRequest.execute({
+        ...params,
+        getUTXOsForAddresses: stateFetcher.getUTXOsForAddresses,
+        getTxsBodiesForUTXOs: stateFetcher.getTxsBodiesForUTXOs,
+      });
+      if (!this.createLedgerSignTxDataRequest.promise) throw new Error('should never happen');
+      const ledgerSignTxDataResp = await this.createLedgerSignTxDataRequest.promise;
+
+      await prepareLedgerConnect(ledgerConnect);
+
+      const ledgerSignTxResp: LedgerSignTxResponse =
+        await ledgerConnect.signTransaction(
+          ledgerSignTxDataResp.ledgerSignTxPayload.inputs,
+          ledgerSignTxDataResp.ledgerSignTxPayload.outputs,
         );
 
-      } else {
-        throw new Error(`LedgerBridge Error: LedgerBridge is undefined`);
-      }
+      await this._prepareAndBroadcastSignedTx(
+        ledgerSignTxResp,
+        params.signRequest.unsignedTx,
+      );
+      Logger.info('Signing on Ledger device is OK');
```

Comment:
Update: users won't be able to connect their Ledger devices in the Yoroi incentivized-testnet release, so the error situation I described won't happen.

Reply:
> If ledger device fails to sign

There are two part involving in making Tx successful on Yoroi side.
1. Signing Tx on the Ledger device.
In case of error, an error is thrown and will directly goto catch block.

2.  _prepareAndBroadcastSignedTx()
As you said its not good error handling.
We need to do two things:
A. Fix error handling.
B. Remove `Logger.info('Signing on Ledger device is OK');` because we already have:
```
this._reset();
Logger.info('SUCCESS: ADA sent using Ledger SignTx');
```
in ` _prepareAndBroadcastSignedTx()`